packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,326 @@
+		assert(opSpaceId3 >= 0 && opSpaceId3 === finalId3);
+
+		assert.strictEqual(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId1,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			localId1,
+		);
+		assert.strictEqual(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId2,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			localId2,
+		);
+		assert.strictEqual(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId3,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			finalId3,
+		);
+	});
+
+	it("eagerly allocates IDs across DDSs using the same compressor", async () => {
+		assert(getIdCompressor(sharedMap1) !== undefined);
+		assert(getIdCompressor(sharedCell1) !== undefined);
+
+		const localId1 = getIdCompressor(sharedMap1).generateCompressedId();
+		assert(localId1 < 0);
+		const localId2 = getIdCompressor(sharedCell1).generateCompressedId();
+		assert(localId2 < 0);
+
+		sharedMap1.set("key", "value");
+		sharedCell1.set("value");
+		await provider.ensureSynchronized();
+
+		const finalId3 = getIdCompressor(sharedMap1).generateCompressedId();
+		assert(finalId3 >= 0);
+		const finalId4 = getIdCompressor(sharedCell1).generateCompressedId();
+		assert(finalId4 >= 0);
+
+		sharedMap1.set("key2", "value2");
+		sharedCell1.set("value2");
+		await provider.ensureSynchronized();
+
+		const opSpaceId1 = getIdCompressor(sharedMap1).normalizeToOpSpace(localId1);
+		const opSpaceId2 = getIdCompressor(sharedCell1).normalizeToOpSpace(localId2);
+		const opSpaceId3 = getIdCompressor(sharedMap1).normalizeToOpSpace(finalId3);
+		const opSpaceId4 = getIdCompressor(sharedCell1).normalizeToOpSpace(finalId4);
+
+		assert(opSpaceId1 >= 0);
+		assert(opSpaceId2 >= 0);
+		assert(opSpaceId3 >= 0 && opSpaceId3 === finalId3);
+		assert(opSpaceId4 >= 0 && opSpaceId4 === finalId4);
+
+		assert.equal(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId1,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			localId1,
+		);
+		assert.equal(
+			getIdCompressor(sharedCell1).normalizeToSessionSpace(
+				opSpaceId2,
+				getIdCompressor(sharedCell1).localSessionId,
+			),
+			localId2,
+		);
+		assert.equal(
+			getIdCompressor(sharedMap1).normalizeToSessionSpace(
+				opSpaceId3,
+				getIdCompressor(sharedMap1).localSessionId,
+			),
+			finalId3,
+		);
+		assert.equal(
+			getIdCompressor(sharedCell1).normalizeToSessionSpace(
+				opSpaceId4,
+				getIdCompressor(sharedCell1).localSessionId,
+			),
+			finalId4,
+		);
+	});
+
+	it("produces Id spaces correctly", async () => {
+		assert(getIdCompressor(sharedMap1) !== undefined);
+		assert(getIdCompressor(sharedMap2) !== undefined);
+		assert(getIdCompressor(sharedMap3) !== undefined);
+
+		const firstId = getIdCompressor(sharedMap1).generateCompressedId();
+		const secondId = getIdCompressor(sharedMap2).generateCompressedId();
+		const thirdId = getIdCompressor(sharedMap2).generateCompressedId();
```

Comment:
Adding container1 / container2 to the name will make it clearer that the operations are happening across containers / clients.

Reply:
I've added these suffixes to the Id names to help clarify what the test is doing.